src/error.rs

```diff
@@ -336,6 +336,8 @@ pub enum ErrorKind {
     UnknownScheme,
     /// DNS lookup failed.
     Dns,
+    /// Insecure request attempted with https only set
+    InsecureRequestHttpsOnly,
```

Comment:
This is a breaking change according to https://github.com/rust-lang/rfcs/blob/master/text/1105-api-evolution.md#major-change-adding-new-variants so it should ideally have bumpped the major version based on semver. To prevent adding new variant becoming breaking change, the enum should be marked as [`#[non_exhaustive]`](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute).

Reply:
An example of downstream breakage: https://github.com/qiniu/rust-sdk/issues/10